File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        string ans = "";
        int a = 0;
        for(int i=0;i<word.length();i++){
            if(word[i]==ch && a<=0){
                ans+=word[i];
                reverse(ans.begin(),ans.end());
                a++;
            }
            else{
                ans+=word[i];
            }
        }
        return ans;
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        StringBuilder ans = new StringBuilder();
        char[] res = word.toCharArray();
        int n = res.length;
        int a = 0;
        for(int i=0;i<n;i++){
            char cha = res[i];
            if(cha==ch && a==0){
                ans.append(cha);
                ans.reverse();
                a++;
            }
            else{
                ans.append(cha);
            }
        }
        return ans.toString();
    }
}